spec/requests/applications_spec.rb

```diff
@@ -22,15 +22,52 @@
     end
 
     it 'allows admin to approve' do
-      app = create(:application)
-      admin_user = create(:user, admin: true)
-      login_as(admin_user)
+      ClimateControl.modify ADMIN_EMAIL: 'identity-admin@example.com' do
+        app = create(:application)
+        admin_user = create(:user, admin: true)
+        deliveries.clear
+        login_as(admin_user)
+
+        mailer = instance_double(ActionMailer::MessageDelivery)
+        allow(UserMailer).to receive(:admin_approved_application).with(anything).and_return(mailer)
+        allow(UserMailer).to receive(:user_approved_application).with(anything).and_return(mailer)
+        allow(mailer).to receive(:deliver_later)
+
+        put users_application_path(app), application: { approved: 'true' }
+
+        expect(response.status).to eq(302) # redirect on success
+        app.reload
+        expect(app.approved?).to eq(true)
+
+        expect(UserMailer).to have_received(:admin_approved_application).with(app)
+        expect(UserMailer).to have_received(:user_approved_application).with(app)
+        expect(mailer).to have_received(:deliver_later).twice
+      end
+    end
+  end
 
-      put users_application_path(app), application: { approved: 'true' }
+  describe 'notifications' do
+    it 'sends email to admin requesting approval' do
+      ClimateControl.modify ADMIN_EMAIL: 'identity-admin@example.com' do
+        user = create(:user)
+        deliveries.clear
+        login_as(user)
+
+        mailer = instance_double(ActionMailer::MessageDelivery)
+        allow(UserMailer).to receive(:admin_new_application).with(anything).and_return(mailer)
+        allow(UserMailer).to receive(:user_new_application).with(anything).and_return(mailer)
+        allow(mailer).to receive(:deliver_later)
+
+        post users_applications_path, application: { name: 'test' }
+
+        app = Application.last
+
+        expect(UserMailer).to have_received(:admin_new_application).with(app)
+        expect(UserMailer).to have_received(:user_new_application).with(app)
+        expect(mailer).to have_received(:deliver_later).twice
 
-      expect(response.status).to eq(302) # redirect on success
-      app.reload
-      expect(app.approved?).to eq(true)
+        expect(response.status).to eq(302)
```

Comment:
Looks great! Perhaps add another expectation here to make sure the app is not yet approved?